### PR TITLE
Working on jsonld to rof

### PR DESCRIPTION
**Pull request by jeremyf**
The primary entry point for the code is https://github.com/ndlib/rof/blob/82368c1426f1b77b176d1dce906f210fca218e95/lib/rof/translators/jsonld_to_rof.rb

## Bumping coverage goal to 94%

@f86d22750c32ff351d42beeb1e3b7c067e985b71

## Exposing public methods to CompareRof

@661553e927c3a188f067b2123df9a615e1bf4a3e

As I'm working on the JSON LD to ROF comparison, I need more detailed
testing to see where I still need to focus the mapping.

Also, leverage the internal instance variable instead of parameter
passing. Last, leverage the Hash#except method of Rails to streamline
a comparison of "everything else".

## Adjusting compare everything else

@73b62999abe6e276db7ed2a1ae74c70e830512d5

Instead of relying on a direct hash comparison, expand the innards of
the hash and compare the arrays as written.

## Translating an ETD's JSON-LD to ROF

@2ffe96bc82a259bbf946ea96b7fd12d12e4c2b67

To orient yourself to the code, review the
`ROF::Translators::JsonldToRof` module.

This is a single tested conversion of one JSON-LD document of an ETD
vs. the Etd node of the ROF document.

Note, this does not yet handle the command line interface of mapping a
JSON-LD document into an ROF document.

DLTP-895

Closes DLTP-906 #Resolve

## Adding JSON-LD test for Document from Curate

@2f2540d5d7aedc5befe6dd9a52674d18bbd95714

## Preparing for 3rd JSON-LD to ROF comparison

@52952fe60a9a73cb08fad2a6cc7530337f880feb

This now handles unescaped unicode hex characters. Huzzah!

## Comparing 4th JSON-LD to ROF (a Patent)

@9f3a8724d2dad2c45e1d6d76f8e56042d1cd3e18

### CODE REVIEW

File: spec/lib/rof/translators/jsonld_to_rof/predicate_object_handler_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+require 'rof/translators/jsonld_to_rof/predicate_object_handler'
+
+module ROF
+  module Translators
+    module JsonldToRof
+      RSpec.describe PredicateObjectHandler do
+        let(:accumulator) { Accumulator.new }
+        let(:rdf_predicate) { RDF::URI.new("http://purl.org/dc/terms/title") }
+        describe '.call' do
+          subject { described_class.call(rdf_predicate, rdf_object, accumulator) }
```
Comment (danhorst) [inline spec/lib/rof/translators/jsonld_to_rof/predicate_object_handler_spec.rb]: I like defining the subject block here and then evaluating it on 15 and 22 _after_ setting a different value for `rdf_object`. This is a pattern I would like to use more.

File: spec/lib/rof/translators/jsonld_to_rof/predicate_object_handler_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          end
+          context 'for an RDF::Node' do
+            let(:rdf_subject) { RDF::Node.new('_b0') }
+            let(:rdf_object) { RDF::Node.new('_b0') }
+            let(:rdf_predicate) { RDF::URI.new("http://www.ndltd.org/standards/metadata/etdms/1.1/") }
+            it 'expands the blank nodes already accumulated values' do
+              statement = RDF::Statement.new(subject: rdf_object, predicate: RDF::URI.new('http://www.ndltd.org/standards/metadata/etdms/1.1/name'), object: RDF::Literal.new('Awesome Sauce'))
+              accumulator.add_blank_node(statement)
+              subject
+              expect(accumulator.to_rof.fetch('metadata').fetch('ms:degree').fetch('ms:name')).to eq(['Awesome Sauce'])
```
Comment (danhorst) [inline spec/lib/rof/translators/jsonld_to_rof/predicate_object_handler_spec.rb]: When I test how a value passes through a function I sometimes set the value to variable but I think using a literal works well here.

File: lib/rof/translators/jsonld_to_rof/accumulator.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          return rof unless @properties
+          rof['properties-meta'] = { "mime-type" => "text/xml" }
+          xml = '<fields>'
+          @properties.each do |node_name, object|
+            xml += "<#{node_name}>#{object}</#{node_name}>"
+          end
+          xml += '</fields>'
+          rof['properties'] = xml
+          rof
+        end
```
Comment (danhorst) [inline lib/rof/translators/jsonld_to_rof/accumulator.rb]: In `lib/rof/compare_rof.rb` individual private methods are marked in the `private : append_properties_to` style. We should probably use one convention or the other.

Comment (jeremyf) [inline lib/rof/translators/jsonld_to_rof/accumulator.rb]: I go back and forth on this. I like to keep related methods close together; I find the interruption of private/public general declarations to be more invasive. However, I moved everything to use the private/public general declaration.

File: lib/rof/compare_rof.rb [hunk 4]
```diff
@@ -3,76 +3,83 @@
...
-      @bendo_rof = bendo_rof
+    def initialize(fedora, bendo, options = {})
+      @fedora = Array.wrap(fedora).first
+      @bendo = Array.wrap(bendo).first
+      @skip_rels_ext_context = options.fetch(:skip_rels_ext_context) { false }
     end
-    attr_reader :fedora_rof, :bendo_rof
+    attr_reader :fedora, :bendo
 
     def error_count
```
Comment (jeremyf) [inline lib/rof/compare_rof.rb]: These are Hashes that are loaded from the ROF source file. They are not recomputed.

**Pull request merged**